supervision/keypoint/annotators.py

```diff
@@ -175,3 +177,109 @@ def annotate(self, scene: ImageType, key_points: KeyPoints) -> ImageType:
+
+
+class VertexLabelAnnotator:
+    """
+    A class for annotating vertex labels on an image using provided detections.
+    """
+
+    def __init__(
+        self,
+        color: Union[Color, List[Color]] = Color.ROBOFLOW,
+        text_color: Color = Color.WHITE,
+        text_scale: float = 0.5,
+        text_thickness: int = 1,
+        text_padding: int = 10,
+        border_radius: int = 0,
+    ):
+        self.border_radius: int = border_radius
+        self.color: Union[Color, List[Color]] = color
+        self.text_color: Color = text_color
+        self.text_scale: float = text_scale
+        self.text_thickness: int = text_thickness
+        self.text_padding: int = text_padding
+
+    @staticmethod
+    def get_text_bounding_box(
+        text: str,
+        font: int,
+        text_scale: float,
+        text_thickness: int,
+        center_coordinates: Tuple[int, int],
+    ) -> Tuple[int, int, int, int]:
+        text_w, text_h = cv2.getTextSize(
+            text=text,
+            fontFace=font,
+            fontScale=text_scale,
+            thickness=text_thickness,
+        )[0]
+        center_x, center_y = center_coordinates
+        return (
+            center_x - text_w // 2,
+            center_y - text_h // 2,
+            center_x + text_w // 2,
+            center_y + text_h // 2,
+        )
+
+    def annotate(
+        self, scene: ImageType, key_points: KeyPoints, labels: List[str] = None
+    ) -> ImageType:
+        font = cv2.FONT_HERSHEY_SIMPLEX
+
+        N, K, _ = key_points.xy.shape
+
+        if N == 0:
+            return scene
+
+        anchors = key_points.xy.reshape(K * N, 2).astype(int)
+        colors = (
+            np.array(self.color * N)
+            if isinstance(self.color, list)
+            else np.array([self.color] * K * N)
+        )
+        labels = np.array(labels * N)
+
+        mask = np.all(anchors != 0, axis=1)
+
+        if np.all(mask == False):
+            return scene
+
+        anchors = anchors[mask]
+        colors = colors[mask]
+        labels = labels[mask]
+
+        xyxy = np.array(
+            [
+                self.get_text_bounding_box(
+                    text=label,
+                    font=font,
+                    text_scale=self.text_scale,
+                    text_thickness=self.text_thickness,
+                    center_coordinates=tuple(anchor),
+                )
+                for anchor, label in zip(anchors, labels)
+            ]
+        )
+
+        xyxy_padded = pad_boxes(xyxy=xyxy, px=self.text_padding)
+
+        for text, color, box, box_padded in zip(labels, colors, xyxy, xyxy_padded):
+            draw_rounded_rectangle(
+                scene=scene,
+                rect=Rect.from_xyxy(box_padded),
+                color=color,
+                border_radius=self.border_radius,
+            )
+            cv2.putText(
+                img=scene,
+                text=text,
+                org=(box[0], box[1] + self.text_padding),
```

Comment:
Assuming you wanted to align keypoint to centers of capital letters, it should be `org=(box[0], box[3])`.

Edit: assuming text thickness of 1. Higher values seem to push the text to the left.

1. Before, 0.5x (default text scale):
![bus_annotated](https://github.com/roboflow/supervision/assets/6500785/704d414a-08f1-444a-a323-b51da41ec893)

2. Before, 1.5x
![bus_annotated](https://github.com/roboflow/supervision/assets/6500785/1ab33004-37c9-44b8-90b2-7725b4e03009)

3. After, 0.5x:
![bus_annotated](https://github.com/roboflow/supervision/assets/6500785/a9958e69-6ebf-4a29-929d-cfb4bb21869d)

4. After, 1.5x
![bus_annotated](https://github.com/roboflow/supervision/assets/6500785/010816e9-c4de-4b49-b852-20ae8240eeba)